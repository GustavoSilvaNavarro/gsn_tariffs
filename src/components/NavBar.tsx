import { Link } from 'react-router';
import ApartmentRoundedIcon from '@mui/icons-material/ApartmentRounded';
import InsertDriveFileRoundedIcon from '@mui/icons-material/InsertDriveFileRounded';
import gsnIcon from '@/assets/edf_logo.png';

export const NavBar = () => {
  return (
    <header className="h-screen flex-[0.2_1_0%] min-w-[280px] bg-[#020712] text-white">
      <nav className="py-4">
        <Link to="/" className="flex items-center justify-center p-4 cursor-pointer mb-2">
          <img src={gsnIcon} alt="Logo" className="w-[20%]" />
          <h3 className="text-3xl ml-3">GSN</h3>
        </Link>

        <div>
          <Link
            to="/utilities"
            className="flex items-center justify-center p-4 transition-all duration-300 ease-in-out cursor-pointer hover:bg-white/10 group-hover:blur-sm hover:!blur-none">
            <ApartmentRoundedIcon />
            <h3 className="text-xl ml-2">Utilities</h3>
          </Link>
          <Link
            to="/tariffs"
            className="flex items-center justify-center p-4 transition-all duration-300 ease-in-out cursor-pointer hover:bg-white/10 group-hover:blur-sm hover:!blur-none">
            <InsertDriveFileRoundedIcon />
            <h3 className="text-xl ml-2">Tariffs</h3>
          </Link>
        </div>
      </nav>
    </header>
  );
};
