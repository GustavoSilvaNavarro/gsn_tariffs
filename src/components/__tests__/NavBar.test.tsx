import { expect } from 'vitest';
import { render, screen, within } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router';
import { userEvent } from '@testing-library/user-event';

import { NavBar } from '../NavBar'; // Adjust the import path if your file structure is different

vi.mock('@/assets/edf_logo.png', () => ({
  default: 'mocked-gsn-logo.png',
}));

describe('NavBar Component tests', () => {
  test('Component should render the data and take a snapshot', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <NavBar />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('Should render the GSN logo and title correctly', () => {
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>,
    );

    const logoElement = screen.getByAltText(/logo/i);

    expect(logoElement).toBeInTheDocument();
    expect(logoElement).toHaveAttribute('src', 'mocked-gsn-logo.png');
    expect(screen.getByText(/gsn/i)).toBeInTheDocument();
  });

  test('Should renders all navigation links with correct `href` attributes', () => {
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>,
    );

    // Find the Home link using its `data-testid` and check its `href`
    const homeLink = screen.getByTestId('cy-home-route');

    expect(homeLink).toBeInTheDocument();
    expect(homeLink).toHaveAttribute('href', '/');

    const utilitiesLink = screen.getByTestId('cy-utilities-route');

    expect(utilitiesLink).toBeInTheDocument();
    expect(utilitiesLink).toHaveAttribute('href', '/utilities');

    const tariffsLink = screen.getByTestId('cy-tariffs-route');
    const tariffsHeading = within(tariffsLink).getByText('Tariffs', { selector: 'h3' });

    expect(tariffsLink).toBeInTheDocument();
    expect(tariffsLink).toHaveAttribute('href', '/tariffs');
    expect(tariffsHeading).toHaveTextContent('Tariffs');
    expect(tariffsLink).toContainElement(tariffsHeading);
  });

  test('Should navigates to the utilities page when the utilities link is clicked', async () => {
    const user = userEvent.setup();
    const UtilitiesPage = () => <div>Utilities Page</div>;

    render(
      <MemoryRouter initialEntries={['/', '/utilities']}>
        <NavBar />
        <Routes>
          <Route path="/utilities" element={<UtilitiesPage />} />
        </Routes>
      </MemoryRouter>,
    );

    const utilitiesLink = screen.getByRole('link', { name: /utilities/i });
    await user.click(utilitiesLink);

    expect(screen.getByText('Utilities Page')).toBeInTheDocument();
  });

  test('Should render Material-UI icons', () => {
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>,
    );

    const icons = screen.getAllByTestId('navbar-mui-icons');

    expect(icons).toHaveLength(2);
  });
});
