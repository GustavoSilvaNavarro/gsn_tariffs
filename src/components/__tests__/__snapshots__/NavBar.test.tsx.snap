// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar Component tests > Component should render the data and take a snapshot 1`] = `
<div>
  <header
    class="h-screen flex-[0.2_1_0%] min-w-[280px] bg-[#020712] text-white"
  >
    <nav
      class="py-4"
    >
      <a
        class="flex items-center justify-center p-4 cursor-pointer mb-2"
        data-discover="true"
        data-testid="cy-home-route"
        href="/"
      >
        <img
          alt="Logo"
          class="w-[20%]"
          src="mocked-gsn-logo.png"
        />
        <h3
          class="text-3xl ml-3"
        >
          GSN
        </h3>
      </a>
      <div>
        <a
          class="flex items-center justify-center p-4 transition-all duration-300 ease-in-out cursor-pointer hover:bg-white/10 group-hover:blur-sm hover:!blur-none"
          data-discover="true"
          data-testid="cy-utilities-route"
          href="/utilities"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="navbar-mui-icons"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 11V5c0-1.1-.9-2-2-2H9c-1.1 0-2 .9-2 2v2H5c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h5c.55 0 1-.45 1-1v-3h2v3c0 .55.45 1 1 1h5c1.1 0 2-.9 2-2v-6c0-1.1-.9-2-2-2zM7 19H5v-2h2zm0-4H5v-2h2zm0-4H5V9h2zm4 4H9v-2h2zm0-4H9V9h2zm0-4H9V5h2zm4 8h-2v-2h2zm0-4h-2V9h2zm0-4h-2V5h2zm4 12h-2v-2h2zm0-4h-2v-2h2z"
            />
          </svg>
          <h3
            class="text-xl ml-2"
          >
            Utilities
          </h3>
        </a>
        <a
          class="flex items-center justify-center p-4 transition-all duration-300 ease-in-out cursor-pointer hover:bg-white/10 group-hover:blur-sm hover:!blur-none"
          data-discover="true"
          data-testid="cy-tariffs-route"
          href="/tariffs"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="navbar-mui-icons"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 2c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8.83c0-.53-.21-1.04-.59-1.41l-4.83-4.83c-.37-.38-.88-.59-1.41-.59zm7 6V3.5L18.5 9H14c-.55 0-1-.45-1-1"
            />
          </svg>
          <h3
            class="text-xl ml-2"
          >
            Tariffs
          </h3>
        </a>
      </div>
    </nav>
  </header>
</div>
`;
